palace/utils/iodata.cpp

```diff
@@ -16,14 +20,131 @@
 namespace palace
 {
 
+namespace
+{
+
+std::stringstream PreprocessFile(const char *filename)
+{
+  // Read configuration file into memory and return as a stringstream.
+  std::string file;
+  {
+    std::ifstream fi(filename);
+    std::stringstream buf;
+    if (!fi.is_open())
+    {
+      MFEM_ABORT("Unable to open configuration file \"" << filename << "\"!");
+    }
+    buf << fi.rdbuf();
+    fi.close();
+    file = buf.str();
+  }
+
+  // Strip C and C++ style comments (//, /* */) using regex. Correctly handles comments
+  // within strings and escaped comment markers (see tinyurl.com/2s3n8dkr).
+  {
+    std::regex rgx(R"((([\"'])(?:(?=(\\?))\3.)*?\2))"
+                   R"(|(\/\*([^*]|[\r\n]|(\*+([^*\/]|[\r\n])))*\*+\/))"
+                   R"(|(\/\/.*))");
+    file = std::regex_replace(file, rgx, "$1");
+  }
+
+  // Also strip whitespace.
+  {
+    std::regex rgx(R"((([\"'])(?:(?=(\\?))\3.)*?\2))"
+                   R"(|(\s+))");
+    file = std::regex_replace(file, rgx, "$1");
+  }
+
+  // Also strip erroneous trailing commas.
+  {
+    std::regex rgx(R"((([\"'])(?:(?=(\\?))\3.)*?\2))"
+                   R"(|,+(?=\s*?[\}\]]))");
+    file = std::regex_replace(file, rgx, "$1");
+  }
+
+  // Perform integer range expansion for arrays ([a - b, c] = [a-b,c] =
+  // [a,a+1,...,b-1,b,c]). The whole file is now one line and arrays have no spaces after
+  // whitespace stripping.
+  std::stringstream output;
+  auto RangeExpand = [](std::string_view str) -> std::string
+  {
+    // Handle the given string which is only numeric with possible hyphens.
+    int num;
+    auto [ptr, ec] = std::from_chars(str.data(), str.data() + str.length(), num);
```

Comment:
Looking at https://en.cppreference.com/w/cpp/utility/from_chars I think the `MFEM_VERIFY` could make use of the `ec` instead
```
MFEM_VERIFY(ec == std::errc(), "Invalid integer conversion in range expansion!");
```
this would also check for overflowing integer values, (which hopefully is a theoretical issue, but who knows).